src/web/transactions.rs

```diff
@@ -1,187 +1,214 @@
+    signatures: SignaturesOf<&TransactionPayload>
+}
+
+impl TransactionBase {
+    fn new<'a, T, U>(hash: HashOf<VersionedSignedTransaction>, payload: TransactionPayload, signatures: T) -> Result<Self>
+    where
+        T: IntoIterator<Item = U>,
+        U: Into<iroha_crypto::Signature> + std::convert::From<&'a SignatureOf<TransactionPayload>>,
+    {
+        Ok(Self {
+            hash: hash,
+            // FIXME https://github.com/hyperledger/iroha/issues/2301
+            // block_hash: Hash::zeroed().into(),
+
+            payload: payload.try_into().wrap_err("Failed to map Payload")?,
+            signatures: signatures
+                .into_iter()
+                .map(Into::<iroha_crypto::Signature>::into)
+                .map(Into::into)
+                .collect(),
+        })
+    }
+}
+
+
+#[derive(Serialize)]
+pub struct CommittedTransactionDTO {
+    #[serde(flatten)]
+    base: TransactionBase,
+}
+
+/// Just as [`CommittedTransactionDTO`], but with rejection reason
+#[derive(Serialize)]
+pub struct RejectedTransactionDTO {
+    #[serde(flatten)]
+    base: TransactionBase,
+    rejection_reason: SerScaleHex<TransactionRejectionReason>,
+}
+
+#[derive(Serialize)]
+pub struct TransactionPayloadDTO {
+    account_id: String,
+    instructions: ExecutableDTO,
+    creation_time: Timestamp,
+    time_to_live_ms:  Option<NonZeroU64>,
+    nonce: Option<NonZeroU32>,
+    metadata: UnlimitedMetadata,
+}
+
+
+
+impl TryFrom<TransactionPayload> for TransactionPayloadDTO {
+    type Error = color_eyre::Report;
+
+    fn try_from(payload: TransactionPayload) -> Result<Self, Self::Error> {
+        Ok(Self {
+            account_id: payload.authority.to_string(),
+            instructions: payload.instructions.into(),
+            creation_time: Timestamp::try_from(payload.creation_time_ms)
+                .wrap_err("Failed to map creation_time")?,
+            time_to_live_ms: payload.time_to_live_ms.try_into()?,
+            nonce: payload.nonce,
+            metadata: payload.metadata,
+        })
+    }
+}
+
+/// Reflection of [`Executable`].
+#[derive(Serialize)]
+#[serde(tag = "t", content = "c")]
+pub enum ExecutableDTO {
+    Instructions(Vec<SerScaleHex< InstructionBox>>),
+    /// WASM binary content is omitted for frontend
+    Wasm,
+}
+
+impl From<Executable> for ExecutableDTO {
+    fn from(value: Executable) -> Self {
+        match value {
+            Executable::Instructions(items) => {
+                Self::Instructions(items.into_iter().map(SerScaleHex).collect())
+            }
+            Executable::Wasm(_) => Self::Wasm,
+        }
+    }
+}
+
+#[get("/{hash}")]
+async fn show(
+    app: web::Data<AppData>,
+    hash: web::Path<HashDeser>,
+) -> Result<web::Json<TransactionDTO>, WebError> {
+    let hash = hash.into_inner().0;
+    let tx = app
+        .iroha_client
+        .request(QueryBuilder::new(FindTransactionByHash::new(HashOf::from_untyped_unchecked(hash)))) // deprecated 

```

Comment:
What is deprecated?

Reply:
The function ```from_untyped_unchecked``` is deprecated. 
![image](https://github.com/soramitsu/iroha2-block-explorer-backend/assets/76839614/732a7df4-871e-41e9-94fe-36fc671e7529)

Reply:
This is the last warning to be taken care of => 
 ```
warning: use of deprecated associated function `iroha_crypto::HashOf::<T>::from_untyped_unchecked`
   --> src/web/transactions.rs:140:21
    |
140 |             HashOf::from_untyped_unchecked(hash),
    |                     ^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: `iroha2_explorer_web` (bin "iroha2_explorer_web") generated 1 warning```

Reply:
So, the documentation string clearly tells you what method to use instead. Try to replace this method with `HashOf::new`.

Reply:
I took a closer look into the types, and indeed, the only way to construct a `FindTransactionByHash` is to use `HashOf<VersionedSignedTransaction>`. The only way to construct it from `Hash` is to use the deprecated method.

I brought that into the discussion with the team today (namely, with @appetrosyan and @mversic). They are going to undeprecate the method soon. The points that were being said:

- Having strong types within the core (i.e. within Iroha codebase) is definitely a good practice, but it shouldn't be mandatory.
- Having a method like `new_untyped_unchecked` is totally fine, although it should be still avoided if possible
- `HashOf` helps to add extra type information to data, but it might not be always the necessary thing.
- While it makes sense to have `HashOf<VersionedSignedTransaction>` in the query structure to handle it in the core, it doesn't make sense for the clients. The clients are external users of Iroha Rust Client API, including Explorer. For example, in our case, we just have some `Hash` whatever it actually belongs to, and we don't _really need_ to specify extra type information by wrapping it into `HashOf<..>`.

The final solution is to simply undeprecate the method back. Therefore, we can ignore the deprecation warning.

Reply:
@horizenight, could we resolve this conversation?